### ISSUE TITLE
Warn for asterisks in strings

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1624]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/sh

ls -laF "/usr/*"
```

#### Here's what shellcheck currently says:
```
Nothing
```
#### Here's what I wanted or expected to see:
Warning that asterisks can't be used in that form.
For example:
```sh
ls "/usr/*"
ls: /usr/*: No such file or directory
```
The finicky part here however is, that this IS valid syntax for at least find. And an asterisks in a string or comment can be valid as well.

So only for cases where the * was intented to do shell globbing is where the warning should be.
